tests/test_builders/test_build_html_tocdepth.py

```diff
@@ -52,9 +54,17 @@ def test_tocdepth(app, cached_etree_parse, fname, path, check, be_found):
     check_xpath(cached_etree_parse(app.outdir / fname), fname, path, check, be_found)
 
 
+def _intradocument_check(nodes: Sequence[Element]) -> None:
+    assert nodes
+    for node in nodes:
+        assert 'href' in node.attrib, "Node is missing an href attribute"
+        msg = f"Node href {node.attrib['href']} does not begin with URI fragment identifier"
+        assert node.attrib['href'].startswith('#'), msg
+
+
 @pytest.mark.parametrize("expect", [
-    (".//li[@class='toctree-l3']/a", '1.1.1. Foo A1', True),
-    (".//li[@class='toctree-l3']/a", '1.2.1. Foo B1', True),
+    (".//li[@class='toctree-l3']/a[.='1.1.1. Foo A1']", _intradocument_check),
+    (".//li[@class='toctree-l3']/a[.='1.2.1. Foo B1']", _intradocument_check),
```

Comment:
Perhaps it would be better to leave these two test cases as-is, and instead to add an additional expectation that all `toctree` links (regardless of text content) are intradocument.

Reply:
Yes, it's may be better. Namely, add a test which checks the '#' of every link in the document.

Reply:
Ah, ok; I began with the `toctree`, but perhaps it would make sense to check the entire document; it does make me wonder whether to add a counter-example of a project that also contains an external hyperlink, though - to make sure that remains unmodified.

Reply:
Yep, I think `tests/roots/test-toctree` could be useful for additional coverage (it includes external links -- they remain valid, as expected, but I'd be marginally more comfortable with this changeset if we test cover that too).